### PR TITLE
Add an "app" mode which lints CSS files that do not describe components

### PR DESCRIPTION
**Pull request by jocelyn-stericker**

### CODE REVIEW

File: test/fixtures/invalid-z-index-2.css [hunk 1]
```diff
@@ -0,0 +1,6 @@
+/** @define InvalidZIndexTwo; use strict */
+
+.InvalidZIndexTwo {
+  --this-has-the-substring-var-but-is-not-a-variable: 2;
```
Comment (michelle) [inline test/fixtures/invalid-z-index-2.css:4]: Haha, clever (:

**Comment by michelle** [comment]
This diff seems to include the diff for your z-index changes--I'm going to just review this one and close the other.

**Comment by michelle** [comment]
So components will always include base app rules?

File: index.js [hunk 2]
```diff
@@ -28,28 +29,39 @@ var COMPOSITION_RE_DIRECTIVE = /@compose (([A-Z][a-zA-Z]+,?[\s]*)+)/;
 
 function conformance(ast, reworkInstance) {
   var initialComment = ast.rules[0].comment;
-  var isDefinition = (initialComment && initialComment.match(/@define/));
+  var isApp = (initialComment && initialComment.match(/@app/));
```
Comment (michelle) [inline index.js]: can probably combine this into a single block for readability:

``` javascript
var isApp, isDefinition, isComponent;
if (initialComment) {
  isApp = ...;
  ...
}
```

File: index.js [hunk 3]
```diff
@@ -28,28 +29,39 @@ var COMPOSITION_RE_DIRECTIVE = /@compose (([A-Z][a-zA-Z]+,?[\s]*)+)/;
...
   var isComponent = (initialComment && initialComment.match(RE_DIRECTIVE));
-  if (!isDefinition) { return; }
-  if (isDefinition && !isComponent) {
+  var isDefinition = (initialComment && initialComment.match(/@define/));
+  if (isApp && (isComponent || isDefinition)) {
```
Comment (michelle) [inline index.js]: IDK if you'd prefer something like this:

``` js
if (isDefinition) {
  if (isComponent) {
    // DO COMPONENT STUFF HERE!
  } else {
    // SHOW WARNINGS HERE
  }

  isApp = true;
}

if (isApp) {
  runBaseTests(rules);
}

```

IDK that the component/app warning is really super necessary. Up to you though!

File: lib/validate-z-indices.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+      }
+
+      if (declaration.property !== 'z-index') {
+        return;
+      }
+
+      var column = declaration.position.start.column;
+      var line = declaration.position.start.line;
+
+      if (declaration.value.indexOf("--") === -1 || declaration.value.indexOf("var(") === -1) {
```
Comment (michelle) [inline lib/validate-z-indices.js]: Why not combine these? `indexOf('var(--')`

File: lib/validate-z-indices.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+module.exports = validateZIndices;
+
+/**
+ * @param {Array} rules
+ */
+
+function validateZIndices(rules) {
+  rules.forEach(function (rule) {
+    if (!isValidRule(rule)) return;
```
Comment (michelle) [inline lib/validate-z-indices.js]: Should we just include `validateRules` before the other `@app` validations?

Comment (michelle) [inline lib/validate-z-indices.js]: (Instead of as part of the component validations)

Comment (jocelyn-stericker) [inline lib/validate-z-indices.js]: Upon closer inspection, `isValidRule` is not required for validation here.

**Comment by michelle** [comment]
LGTM! :sheep:

**Comment by jocelyn-stericker** [comment]
I've squashed my commits, however I do not have write access to this repository.

**Comment by michelle** [comment]
Thanks for the squash, will merge.

**Pull request merged**